From 41ed4217a1c8db49e4919f0ba43aef1e0ae90ecd Mon Sep 17 00:00:00 2001
From: Smita Gaikwad <smitag@nvidia.com>
Date: Mon, 18 Jun 2012 03:17:34 -0400
Subject: [PATCH 10/40] Implemented Range attribute parsing

SDP Parsers should be tolerant enough to parse the non-standard
a=range attribute in the Session Description. Implemented the
parsing of the Range attribute.

Bug 845202

Integration for following commit:
http://git-master/r/40012

Change-Id: Idcec4d8eca456dea0686e94fd3019e7e657c3051
Reviewed-on: http://git-psac/r/271
Tested-by: Smita Gaikwad <smitag@nvidia.com>
Reviewed-by: Suresh Choudhary <sureshc@nvidia.com>
Reviewed-by: Aniruddha Chourasia <aniruddhac@nvidia.com>
Reviewed-by: Parag Mehendale <pmehendale@nvidia.com>
Reviewed-by: Zhijun He <zhhe@nvidia.com>
---
 media/libstagefright/rtsp/ASessionDescription.cpp |   10 +++++++---
 1 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/media/libstagefright/rtsp/ASessionDescription.cpp b/media/libstagefright/rtsp/ASessionDescription.cpp
index a9b3330..c6b732f 100644
--- a/media/libstagefright/rtsp/ASessionDescription.cpp
+++ b/media/libstagefright/rtsp/ASessionDescription.cpp
@@ -259,12 +259,16 @@ bool ASessionDescription::getDurationUs(int64_t *durationUs) const {
         return false;
     }
 
-    if (strncmp(value.c_str(), "npt=", 4)) {
+    float from, to;
+    uint8_t offset = 0;
+    if (!strncmp(value.c_str(), "npt", 3)) {
+        // Demiliter in the Range Type field can be npt: npt=
+        offset = 4;
+    } else if ((!strncmp(value.c_str(), "smpte", 5)) || (!strncmp(value.c_str(), "clock", 5))) {
         return false;
     }
 
-    float from, to;
-    if (!parseNTPRange(value.c_str() + 4, &from, &to)) {
+    if (!parseNTPRange(value.c_str() + offset, &from, &to)) {
         return false;
     }
 
-- 
1.7.1

